File: src/java1/q4.java
//Calculate the number & Percentage Of Lowercase Letters,Uppercase Letters, Digits And Other Special Characters In A String
package java1;

public class q4 {
    public static void main(String[] args) {
        int uCount = 0;
        int lCount = 0;
        int digitCount = 0;
        int specialCount = 0;
        String st = "UUUU World ** 13";
        for (int i = 0; i < st.length(); i++) {
            char ch = st.charAt(i);
            if ((Character.isUpperCase(ch))) {
                uCount++;
            } else if ((Character.isLowerCase(ch))) {
                lCount++;

            } else if (Character.isDigit(ch)) {
                digitCount++;
            } else {
                specialCount++;
            }

        }

        double length = st.length();
        double upperPercent = (uCount / length) * 100;
        double lowerPercent = (lCount / length) * 100;
        double digitPercent = (digitCount / length) * 100;
        double specialPercent = (specialCount/length)*100;
        System.out.println("Upper Percent:" + upperPercent + "\n" +"Lower Percent:" + lowerPercent
                + "\n" + "Digit Percent:" + digitPercent + "\n" + "Special Percent:" +specialCount );

    }


}

File: Exercise1/q4.kt
package AssesmentOne

//Calculate the number & Percentage Of Lowercase Letters,Uppercase Letters, Digits And Other Special Characters In A String.
fun main() {
    println("Enter the String:")
    var str : String = readLine().toString()
    var nL = 0
    var nU = 0
    var nD = 0
    var specialChar = 0
    val len : Int = str.length
    for(i in 0 until len){
        when {
            str[i].isUpperCase() -> nU++
            str[i].isLowerCase() -> nL++
            str[i].isDigit() -> nD++
            else -> specialChar++
        }
    }

    var uppercaseLetterPercentage = (nU * 100) / len
    var lowercaseLetterPercentage = (nL * 100) / len
    var specialCharacterLetterPercentage = (specialChar * 100) / len

    println("length is : $len")
    println("Upper case Letters are : $uppercaseLetterPercentage%")
    println("Lower case Letters are : $lowercaseLetterPercentage%")
    println("Special case Letters are : $specialCharacterLetterPercentage%")

}